core/components/molecules/dialog/dialog.js

```diff
@@ -34,75 +35,174 @@ const createButtonForAction = (action, index) => {
+              {...Automation('dialog.body')}
+            >
+              {props.children}
+            </DialogBody>
+
+            {props.actions && (
+              <DialogFooter {...Automation('dialog.footer')}>
+                <ButtonGroup>{props.actions.map(createButtonForAction)}</ButtonGroup>
+              </DialogFooter>
+            )}
+          </DialogBox>
+        </FocusTrap>
+      </Overlay>
+    )
+  }
+}
+
+const DialogBox = styled.div`
   position: relative;
-  overflow: hidden;
-  width: ${props => props.width}px;
-  background: ${colors.base.white};
+  max-height: calc(100vh - (${spacing.xlarge} * 2));
+  display: flex;
+  flex-direction: column;
+  background-color: ${colors.base.white};
   border-radius: 3px;
   box-shadow: 0 5px 15px rgba(0, 0, 0, 0.5);
 `
 
-const DialogTitleBar = styled.div`
-  display: flex;
+const DialogClose = styled.div`
+  position: absolute;
+  top: 0;
+  right: 0;
+  z-index: 1;
+
+  /* Overwirtes the color of the icons */
+  ${Button.Element} {
+    &,
+    i,
+    svg,
+    path {
+      color: ${colors.base.black};
+      fill: ${colors.base.black};
+    }
+  }
+`
+
+const DialogHeader = styled.header`
   position: relative;
-  overflow: hidden;
-  padding: ${spacing.small};
-  padding-bottom: 0;
+  padding: ${spacing.small} ${spacing.large} ${spacing.xsmall} ${spacing.large};
   color: ${colors.text.default};
-  font-weight: ${fonts.weight.medium};
+  word-break: break-word;
+  white-space: pre-wrap;
+  word-wrap: break-word;
   text-align: center;
-  span {
-    flex: 1;
-  }
-  & ${Icon.Element} {
-    flex: none;
-    cursor: pointer;
-  }
+`
+
+const DialogTitle = styled(Heading)`
+  font-weight: ${fonts.weight.medium};
+  font-size: ${fonts.size.default};
+  margin: 0;
 `
 
 const DialogBody = styled.div`
-  padding: ${spacing.medium} ${spacing.large};
+  padding: ${spacing.small} ${spacing.medium} ${spacing.large} ${spacing.medium};
+  flex: 1 1 auto;
+  overflow-y: auto;
+  -webkit-overflow-scrolling: touch;
+  overscroll-behavior: contain;
+  word-break: break-word;
+  word-wrap: break-word;
+
+  /* Clears the margin of the last item of the body */
+  > * {
+    margin-bottom: 0;
+  }
 `
 
-const DialogFooter = styled.div`
+const DialogFooter = styled.footer`
   display: flex;
+  flex: 0 0 auto;
   justify-content: center;
   padding: ${spacing.small};
   border-top: 1px solid ${colors.base.grayLight};
 `
 
 Dialog.Action = DialogAction
-Dialog.Element = DialogElement
-
+Dialog.Element = DialogBox
 Dialog.propTypes = {
+  /** Buttons that will be shown on the dialog's footer */
   actions: PropTypes.arrayOf(
     PropTypes.oneOfType([PropTypes.instanceOf(DialogAction), PropTypes.element])
-  ).isRequired,
-  title: PropTypes.string.isRequired,
-  width: PropTypes.number,
-  onClose: PropTypes.func
+  ),
+  /** Dialog's header title */
+  title: PropTypes.string,
+  /** Dialog's header title heading element */
+  titleElement: PropTypes.oneOf([1, 2, 3, 4]),
```

Comment:
Needs a better name.

Looking at the prop name, I didn't realise this is what <h*> tag will be used. I'd recommend using a prop that matches the underlying `Heading` component: https://auth0-cosmos.now.sh/docs/#/component/heading

```js
/** Title in header */
title: PropTypes.string,
/** size of `Heading` used in title */
titleSize: PropTypes.oneOf([1, 2, 3, 4])
```

`headingSize` could also work but it wouldn't feel related to `title` then

Reply:
The ultimate goal is not to relate the heading element to the size. So we went with this approach to be covered if in the future we make the heading element independent of the size, this is one less prop to deprecate.

Reply:
Got it! This isn't related to size at all, but only the h tag.

To make this obvious, can we change the accepted values to 

```js
titleElement: PropTypes.oneOf(['h1', 'h2', 'h3', 'h4'])
```

Reply:
Moving this conversation to the conversation tab